internal/ui/src/components/date-time/components/calendar.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+// biome-ignore lint: React in this context is used throughout, so biome will change to types because no APIs are used even though React is needed.
+import * as React from "react";
+import { type CaptionProps, DayPicker, IconLeft, IconRight, useNavigation } from "react-day-picker";
+
+import { format } from "date-fns";
+import { cn } from "../../../lib/utils";
+import { buttonVariants } from "../../button";
+
+export type CalendarProps = React.ComponentProps<typeof DayPicker>;
+function CustomCaptionComponent(props: CaptionProps) {
+  const { goToMonth, nextMonth, previousMonth } = useNavigation();
+  return (
+    <div className="flex flex-row w-full">
+      <button disabled={!previousMonth} onClick={() => previousMonth && goToMonth(previousMonth)}>
+        <IconLeft className="text-gray-12 size-2" />
+      </button>
+      <div className="w-full text-center text-gray-12 text-xs">
+        {format(props.displayMonth, "MMMM yyy")}
+      </div>
+      <button disabled={!nextMonth} onClick={() => nextMonth && goToMonth(nextMonth)}>
+        <IconRight className="text-gray-12 size-2" />
+      </button>
+    </div>
+  );
+}
+function CalendarBase({ className, classNames, showOutsideDays = true, ...props }: CalendarProps) {
```

Comment:
Btw, we don't have to call this `CalendarBase` I did it in the specs, because we'll be building our calendar on this base.